mixer/test/client/pilotplugin/pilotplugin_test.go

```diff
@@ -0,0 +1,377 @@
+	"github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
+	"github.com/envoyproxy/go-control-plane/envoy/api/v2/listener"
+	"github.com/envoyproxy/go-control-plane/envoy/api/v2/route"
+	hcm "github.com/envoyproxy/go-control-plane/envoy/config/filter/network/http_connection_manager/v2"
+	discovery "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v2"
+	"github.com/envoyproxy/go-control-plane/pkg/cache"
+	xds "github.com/envoyproxy/go-control-plane/pkg/server"
+	"github.com/envoyproxy/go-control-plane/pkg/util"
+	"google.golang.org/grpc"
+
+	meshconfig "istio.io/api/mesh/v1alpha1"
+	"istio.io/istio/mixer/test/client/env"
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pilot/pkg/networking/plugin"
+	"istio.io/istio/pilot/pkg/networking/plugin/mixer"
+	pilotutil "istio.io/istio/pilot/pkg/networking/util"
+)
+
+const envoyConf = `
+admin:
+  access_log_path: {{.AccessLogPath}}
+  address:
+    socket_address:
+      address: 127.0.0.1
+      port_value: {{.Ports.AdminPort}}
+node:
+  id: id
+  cluster: unknown
+dynamic_resources:
+  lds_config: { ads: {} }
+  ads_config:
+    cluster_names: ["xds"]
+static_resources:
+  clusters:
+  - name: xds
+    http2_protocol_options: {}
+    connect_timeout: 5s
+    type: STATIC
+    hosts:
+    - socket_address:
+        address: 127.0.0.1
+        port_value: {{.Ports.DiscoveryPort}}
+  - name: "outbound|||svc.ns3"
+    connect_timeout: 5s
+    type: STATIC
+    hosts:
+    - socket_address:
+        address: 127.0.0.1
+        port_value: {{.Ports.ServerProxyPort}}
+  - name: "inbound|||backend"
+    connect_timeout: 5s
+    type: STATIC
+    hosts:
+    - socket_address:
+        address: 127.0.0.1
+        port_value: {{.Ports.BackendPort}}
+  - name: "outbound|9091||mixer_server"
+    http2_protocol_options: {}
+    connect_timeout: 5s
+    type: STATIC
+    hosts:
+    - socket_address:
+        address: 127.0.0.1
+        port_value: {{.Ports.MixerPort}}
+`
+
+const checkAttributesOkGet = `
+{
+  "connection.mtls": false,
+  "context.protocol": "http",
+  "context.reporter.kind": "inbound",
+  "context.reporter.uid": "kubernetes://pod1.ns2",
+  "destination.ip": "[0 0 0 0 0 0 0 0 0 0 255 255 127 0 0 1]",
+  "destination.port": "*",
+  "destination.namespace": "ns2",
+  "destination.uid": "kubernetes://pod1.ns2",
+  "destination.service": "svc.ns3",
+  "destination.service.host": "svc.ns3",
+  "destination.service.name": "svc",
+  "destination.service.namespace": "ns3",
+  "destination.service.uid": "svcuid",
+  "source.uid": "kubernetes://pod2.ns2",
+  "request.headers": {
+     ":method": "GET",
+     ":path": "/echo",
+     ":authority": "*",
+     "x-forwarded-proto": "http",
+     "x-request-id": "*"
+  },
+  "request.host": "*",
+  "request.path": "/echo",
+  "request.time": "*",
+  "request.useragent": "Go-http-client/1.1",
+  "request.method": "GET",
+  "request.scheme": "http"
+}
+`
+const reportAttributesOkOutbound = `
```

Comment:
@douglas-reid these are real attributes as coming from pilot generated config and envoy. Let me know if you spot anything wrong...